src/DurableTask.Core/TaskOrchestrationContext.cs

```diff
@@ -571,12 +571,16 @@ public void HandleEventWhileSuspended(HistoryEvent historyEvent)
         public void HandleExecutionSuspendedEvent(ExecutionSuspendedEvent suspendedEvent)
         {
             this.IsSuspended = true;
-            
+
+            //When the orchestrator is suspended, a task could potentially be added to the orchestratorActionsMap.
+            //This could lead to the task being executed repeatedly without completion until the orchestrator is resumed.
+            //To prevent this scenario, check if orchestratorActionsMap is empty before proceeding.
             if (this.orchestratorActionsMap.Any())
             {
-                foreach(var pair in this.orchestratorActionsMap)
+                //If not, store its contents to a temporary dictionary to allow processing of the task later when orchestrator resumes.
```

Comment:
nit - as above, let's add spaces between "//" and the text
```suggestion
                // If not, store its contents to a temporary dictionary to allow processing of the task later when orchestrator resumes.
```